changelog.d/13819.misc

```diff
@@ -0,0 +1 @@
+complement: init postgres DB directly inside the target image instead of the base postgres image to fix building using Buildah.
```

Comment:
I understand this is a `misc` internal fix for Complement testing. But I have no idea what this change is doing.

What is Buildah? -> https://buildah.io/

What are we using https://buildah.io/ for?

What was actually failing before? Was it an error? Was it bugged behavior because data in the volume wasn't there?

The description also makes me scratch my head:

> Doing so in the base postgres image doesn't work with buildah because changes in a declared VOLUME in the Dockerfile is supposed to be discarded, cf https://docs.docker.com/engine/reference/builder/#volume

---

Going through all of these steps I can sorta piece out that the `ALTER USER` and `CREATE DATABASE` pieces of the Docker build which end up persisted to the volume were being discarded (as [mentioned in the Docker docs](https://docs.docker.com/engine/reference/builder/#notes-about-specifying-volumes)) and those are crucial for things to work.

Why does this work before work outside of Buildah (our Complement tests seem to work in CI for example) when it seems to be a Docker behavior itself?

Reply:
I guess https://github.com/matrix-org/synapse/pull/13867 tries to comment about how this works now.

But the PR descriptions and context is completely missing here.

Reply:
We don't use buildah in the project, but I do use it on my machine because I don't have Docker there.

> Why did this work before work outside of Buildah (our Complement tests seem to work in CI for example) when it seems to be a Docker behavior itself?

Because it seems Docker doesn't respect its doc :) Perhaps for legacy reasons.

Reply:
I've edited the PR description to add some context.

Reply:
Thanks for the PR description update @MatMaul! It's a great summary of all of this 💯